### ISSUE TITLE
Code Inspections setting double scrollbars

### ISSUE THREAD

**Issue by bclothier** [rubberduck-vba/Rubberduck#4608]
There are 2 vertical scrollbars for the code inspections setting pane:
![image](https://user-images.githubusercontent.com/2367644/49521211-8f590d80-f86a-11e8-91eb-6b73e43a8ee9.png)

Only one is really needed.

**Comment by retailcoder** [comment]
This is a design dilemma: either...

 - we have a vertical scrollbar for the "inspection severities", because the height of that grouping grid is constrained so as to not bury the other settings on the page, while keeping the most-used part of the page in a prominent position.
 - there's one scrollbar and users only see "inspection severites" at a glance, because the height of that grouping grid is no longer restrained.
 - inspection severities moves underneath misc. and whitelisted identifiers, making the most-used part of the page not as prominent as it should be.

**Comment by Vogel612** [comment]
There's another design issue there. "Import" and "Export" placement implies that they only apply to the Inspection severity settings, when they don't (or shouldn't).

**Comment by bclothier** [comment]
Typically, I'd make the grid anchored so that it grows/shrink with the dialog, keeping the header and footer constant in their height. Furthermore, we can specify a min height (both at the dialog level and at the pane level) to prevent the grid from being shrunk to nothing. That way, we only need one scrollbar which is the inner scrollbar.

**Comment by retailcoder** [comment]
@bclothier no, the "inner" scrollbar belongs to the severities grouping grid. but the inspection settings page is taller than the settings dialog - the "outer" scrollbar belongs to the settings page. The outer scrollbar can't be removed without massively rearranging everything on that page.

**Comment by bclothier** [comment]
My point is that there's no reason to have the settings page be taller than the settings dialog initially. When a dialog opens, everything should be visible without any resizing. In this case, we have a grid which may warrant more screen real estate so we allow resizing to bigger size but not smaller. As a rule, we never should see a double scrollbar. Anchoring the grid, which is of dynamic height helps us avoid that dilemma.

**Comment by retailcoder** [comment]
Except, the settings dialog is completely resizeable. It *must* have a vertical scrollbar.

**Comment by retailcoder** [comment]
How about redesigning the inspection settings page, and make it have "tabs", kind of like the upcoming references dialog? that way severities could use the entire page estate without hindering discoverability of other inspection settings.

**Comment by bclothier** [comment]
The problem with tabs is that the grouping names are quite wordy and would take up lot of screen horizontally. Also, we already have a panel on the left, so adding tabs/panels strikes me as a variation of the double scrollbar problem we want to avoid. It would also complicate the search by name which can be used to yield a nice list across all grouping (assuming they have not chosen to use the grouping dropdown to filter the results). 

The outside scrollbar will not be necessary if we enforce a minimum size on both the dialog and the setting page and make the grid anchored so that it grows/shrinks when the dialog resizes.

**Comment by retailcoder** [comment]
And what happens when the *whitelisted identifiers* list contains more items than can fit the allotted space?
